controllers/cinder_controller.go

```diff
@@ -782,27 +791,32 @@ func (r *CinderReconciler) reconcileNormal(ctx context.Context, instance *cinder
 			Log.Info(fmt.Sprintf("Deployment %s successfully reconciled - operation: %s", instance.Name, string(op)))
 		}
 
-		// Mirror CinderVolume status' ReadyCount to this parent CR
-		// TODO: Somehow this status map can be nil here despite being initialized
-		//       in the Reconcile function above
-		if instance.Status.CinderVolumesReadyCounts == nil {
-			instance.Status.CinderVolumesReadyCounts = map[string]int32{}
-		}
-		instance.Status.CinderVolumesReadyCounts[name] = cinderVolume.Status.ReadyCount
-
-		// If this cinderVolume is not IsReady, mirror the condition to get the latest step it is in.
-		// Could also check the overall ReadyCondition of the cinderVolume.
-		if !cinderVolume.IsReady() {
-			c = cinderVolume.Status.Conditions.Mirror(cinderv1beta1.CinderVolumeReadyCondition)
-			// Get the condition with higher priority for volumeCondition.
-			volumeCondition = condition.GetHigherPrioCondition(c, volumeCondition).DeepCopy()
+		// Only if the generation of the CinderVolume is the same as the observed generation, can we mirror the condition and ready count
```

Comment:
Another instance that might want to repeat L691.

Reply:
Done.